code/modules/crafting/craft.dm

```diff
@@ -1,24 +1,35 @@
 /datum/personal_crafting
 	var/busy
 	var/viewing_category = 1 //typical powergamer starting on the Weapons tab
-	var/list/categories = list(CAT_WEAPON,
-				CAT_AMMO,
+	var/viewing_subcategory = 1
+	var/list/categories = list(
+				CAT_WEAPONRY,
 				CAT_ROBOT,
 				CAT_MISC,
 				CAT_PRIMAL,
-				CAT_BREAD,
-				CAT_BURGER,
-				CAT_CAKE,
-				CAT_EGG,
-				CAT_MEAT,
-				CAT_MISCFOOD,
-				CAT_PASTRY,
-				CAT_PIE,
-				CAT_PIZZA,
-				CAT_SALAD,
-				CAT_SANDWICH,
-				CAT_SOUP,
-				CAT_SPAGHETTI)
+				CAT_FOOD)
+	var/list/subcategories = list(
```

Comment:
Right, but then you'd still need another list to map from integers to categories,  such as:

var/list/categorynames = list(
0 = CAT_NONE,
1 = CAT_WEAPONRY,
2 = CAT_ROBOT,
 ...
15 = CAT_SPAGHETTI)

And any time you'd want to change it you'd have to rewrite the entire categorynames array, since if the associative list is indexed by numbers, you'd have to advance all the subcategories by 1 if you added a new main category.